webpack/assets/javascripts/react_app/redux/middlewares/IntervalMiddleware/IntervalMiddleware.js

```diff
@@ -0,0 +1,36 @@
+import { START_INTERVAL, STOP_INTERVAL } from './IntervalConstants';
+import { selectIntervalID } from './IntervalSelectors';
+import { amendActionsPayload } from '../common/helpers';
+import {
+  registeredIntervalException,
+  unregisteredIntervalException,
+} from './IntervalHelpers';
+
+export const IntervalMiddleware = store => next => action => {
+  const { type, payload: { key, method, interval, args = [] } = {} } = action;
+  const state = store.getState();
+
+  if (type === START_INTERVAL) {
+    const intervalAlreadyExist = !!selectIntervalID(state, key);
+
+    if (intervalAlreadyExist) {
+      throw registeredIntervalException(key);
+    }
+
+    method(); // force the method to run for the first time.
```

Comment:
Very confused by the name of `method` here.  I think maybe it's the whole point of the middleware, some callback that you want to get called every xx milliseconds?  I think maybe `callback` would be a better name?  Or the MDN docs for `setInterval` just call it `func`.

Reply:
same as the comment above about `args`, I tried to stick with the `MDN` docs for `setInterval`,
while they use `func` which is a reserved word in JS, I decided to use `method`.